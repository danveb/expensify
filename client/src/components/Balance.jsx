import { useContext } from "react"; 
import { GlobalContext } from "../context/GlobalState";

const Balance = () => {
    // useContext
    const context = useContext(GlobalContext); 
    // destructure transactions from context
    const { transactions } = context; 

    const amounts = transactions.map((transaction) => transaction.amount); 
    const total = amounts.reduce((acc, item) => (acc += item), 0).toFixed(2);  

    // convert date object 
    const date = new Date().toISOString().split("T")[0]; 

    return (
        <>
            <h4>{`Your Balance as of ${date}`}</h4>
            <h1>$ {total}</h1>
        </>
    )
}

export default Balance