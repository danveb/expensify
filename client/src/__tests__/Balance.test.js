import { render } from "@testing-library/react";
import Balance from "../components/Balance"; 

describe("Balance component", () => {
    // smoke test
    test("renders Balance component", () => {
        render(<Balance />); 
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<Balance />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText
    test("displays correct date", () => {
        const date = new Date().toISOString().split("T")[0]; 
        const { getByText } = render(<Balance />); 
        const text = getByText(`Your Balance as of ${date}`, { exact: true }); 
        expect(text).toBeInTheDocument(); 
    });

    // getByText
    test("displays correct total", () => {
        const { getByText } = render(<Balance />); 
        const totalBalance = getByText(`$`, { exact: false }); 
        expect(totalBalance).toBeInTheDocument(); 
    });
});