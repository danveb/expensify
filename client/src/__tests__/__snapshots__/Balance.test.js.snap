// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance component matches snapshot 1`] = `
<DocumentFragment>
  <h4>
    Your Balance as of 2022-10-12
  </h4>
  <h1>
    $ 2350.00
  </h1>
</DocumentFragment>
`;
